From df3a18a6e24caf21cebee435b351978c8b4060d2 Mon Sep 17 00:00:00 2001
From: Walter Dal Mut <walter.dalmut@corley.it>
Date: Wed, 26 Aug 2026 11:16:52 +0200
Subject: [PATCH 08/74] ARM: dts: rockchip: rk3506: Fix up pinctrl dtsi for the
 arch/arm tree

Two build fixes for the rk3506 pinctrl dtsi as posted:

rockchip-pinconf.dtsi lives only under arch/arm64/boot/dts/rockchip, so
the plain #include "rockchip-pinconf.dtsi" does not resolve from
arch/arm. Use the cross-architecture include form already used by the
sibling file, arch/arm/boot/dts/rockchip/rv1126-pinctrl.dtsi.

rk3506-pinctrl.dtsi references pcfg_input_enable_pull_down eight times,
but that node was never added to rockchip-pinconf.dtsi. Add it. It is
marked /omit-if-no-ref/, so no other board's dtb changes.

Signed-off-by: Walter Dal Mut <walter.dalmut@corley.it>

(cherry picked from commit c7207f45b927c9967e85e60086fb66f46092b9a3)

diff --git a/arch/arm/boot/dts/rockchip/rk3506-pinctrl.dtsi b/arch/arm/boot/dts/rockchip/rk3506-pinctrl.dtsi
index f81dca010..d463ebd12 100644
--- a/arch/arm/boot/dts/rockchip/rk3506-pinctrl.dtsi
+++ b/arch/arm/boot/dts/rockchip/rk3506-pinctrl.dtsi
@@ -4,7 +4,7 @@
  */
 
 #include <dt-bindings/pinctrl/rockchip.h>
-#include "rockchip-pinconf.dtsi"
+#include <arm64/rockchip/rockchip-pinconf.dtsi>
 
 /*
  * This file is auto generated by pin2dts tool, please keep these code
diff --git a/arch/arm64/boot/dts/rockchip/rockchip-pinconf.dtsi b/arch/arm64/boot/dts/rockchip/rockchip-pinconf.dtsi
index b0475b7c6..79af18da2 100644
--- a/arch/arm64/boot/dts/rockchip/rockchip-pinconf.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rockchip-pinconf.dtsi
@@ -367,6 +367,12 @@ pcfg_pull_none_drv_level_5_smt: pcfg-pull-none-drv-level-5-smt {
 		input-schmitt-enable;
 	};
 
+	/omit-if-no-ref/
+	pcfg_input_enable_pull_down: pcfg-input-enable-pull-down {
+		input-enable;
+		bias-pull-down;
+	};
+
 	/omit-if-no-ref/
 	pcfg_output_high: pcfg-output-high {
 		output-high;
